### PR TITLE
Fixes 1971: Task info list filtering and new fields

### PR DESCRIPTION
**Pull request by rverdile**
## Summary
This adds 3 new fields to the task fetch and list API responses: type, repository name, repository (config) UUID. It also adds filtering by type and repository UUID to the task list endpoint.
 
## Testing steps
1. Create a repository with snapshot enabled so that a couple of tasks are produced.
2. List the tasks without filtering, filtering by repository UUID, and filtering by name. Check for new fields.
3. Fetch a task and check for new fields

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-1971

File: pkg/dao/task_info.go [hunk 1]
```diff
@@ -20,11 +36,16 @@ func GetTaskInfoDao(db *gorm.DB) TaskInfoDao {
...
-	result := t.db.Where("id = ? AND org_id = ?", id, orgId).First(&taskInfo)
+func (t taskInfoDaoImpl) Fetch(orgID string, id string) (api.TaskInfoResponse, error) {
+	taskInfo := models.TaskInfoRepositoryConfiguration{}
 	taskInfoResponse := api.TaskInfoResponse{}
 
+	result := t.db.Table(taskInfo.TableName()+" AS t ").
+		Select(JoinSelectQuery).
+		Joins("INNER JOIN repositories r on t.repository_uuid = r.uuid").
+		Joins("INNER JOIN repository_configurations rc on r.uuid = rc.repository_uuid").
+		Where("t.id = ? AND t.org_id = ?", id, orgID).First(&taskInfo)
```
Comment (jlsherrill) [inline pkg/dao/task_info.go:47]: by doing an inner join, you exclude any tasks that don't have an associated repository or repository_configuration.  We probably want to do a Left join, to include those even if the resulting select returns a null repo_config_uuid/name.

**Comment by swadeley** [comment]
Hi

I ran apigen.

I created a repo:

```
Out[4]: 
{'account_id': '0369233',
 'distribution_arch': 'any',
 'distribution_versions': ['any'],
 'failed_introspections_count': 0,
 'gpg_key': '',
 'last_introspection_error': '',
 'last_introspection_time': '',
 'last_success_introspection_time': '',
 'last_update_introspection_time': '',
 'metadata_verification': False,
 'name': 'fedoratest-1',
 'org_id': '3340851',
 'package_count': 0,
 'snapshot': True,
 'status': 'Pending',
 'url': 'https://stephenw.fedorapeople.org/multirepos/5/repo01/',
 'uuid': '54e92977-9c58-43de-be31-bcacc92ce721'}
```


Looks good:

```
In [5]: app.content_sources.rest_client.tasks_api.list_tasks()
2023-08-17 09:32:04.557 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, request_id=XZilDYrmQ2jGx0bZ11iia5bwciCYanWt, params=[]
Out[5]: 
{'data': [{'created_at': '2023-08-17T08:23:23Z',
           'ended_at': '2023-08-17T08:29:22Z',
           'error': '',
           'org_id': '3340851',
           'repository_name': 'fedoratest-1',
           'repository_uuid': '54e92977-9c58-43de-be31-bcacc92ce721',
           'status': 'completed',
           'type': 'introspect',
           'uuid': '48bda3b4-aa25-4183-bc2e-a49009624f89'},
          {'created_at': '2023-08-17T08:23:23Z',
           'ended_at': '2023-08-17T08:29:32Z',
           'error': '',
           'org_id': '3340851',
           'repository_name': 'fedoratest-1',
           'repository_uuid': '54e92977-9c58-43de-be31-bcacc92ce721',
           'status': 'completed',
           'type': 'snapshot',
           'uuid': '674b6996-8e8d-45b3-9532-e60b1c9257e3'}],
 'links': {'first': '/api/content-sources/v1/tasks/?limit=100&offset=0',
           'last': '/api/content-sources/v1/tasks/?limit=100&offset=0'},
 'meta': {'count': 2, 'limit': 100, 'offset': 0}}

```

**Comment by swadeley** [comment]
Test filter for tasks:

```
In [9]: app.content_sources.rest_client.tasks_api.list_tasks.attribute_map
Out[9]: 
{'offset': 'offset',
 'limit': 'limit',
 'status': 'status',
 'type': 'type',
 'repository_uuid': 'repository_uuid'}
```

By UUID of the repo:

```
In [10]: app.content_sources.rest_client.tasks_api.list_tasks(repository_uuid='54e92977-9c58-43de-be31-bcacc92ce721')
2023-08-17 10:11:22.121 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, request_id=JXHeBcq9kMdBS5v1kbUZgtxfkUQK4goZ, params=[('repository_uuid', '54e92977-9c58-43de-be31-bcacc92ce721')]
Out[10]: 
{'data': [{'created_at': '2023-08-17T08:23:23Z',
           'ended_at': '2023-08-17T08:29:22Z',
           'error': '',
           'org_id': '3340851',
           'repository_name': 'fedoratest-1',
           'repository_uuid': '54e92977-9c58-43de-be31-bcacc92ce721',
           'status': 'completed',
           'type': 'introspect',
           'uuid': '48bda3b4-aa25-4183-bc2e-a49009624f89'},
          {'created_at': '2023-08-17T08:23:23Z',
           'ended_at': '2023-08-17T08:29:32Z',
           'error': '',
           'org_id': '3340851',
           'repository_name': 'fedoratest-1',
           'repository_uuid': '54e92977-9c58-43de-be31-bcacc92ce721',
           'status': 'completed',
           'type': 'snapshot',
           'uuid': '674b6996-8e8d-45b3-9532-e60b1c9257e3'}],
 'links': {'first': '/api/content-sources/v1/tasks/?limit=100&offset=0&repository_uuid=54e92977-9c58-43de-be31-bcacc92ce721',
           'last': '/api/content-sources/v1/tasks/?limit=100&offset=0&repository_uuid=54e92977-9c58-43de-be31-bcacc92ce721'},
 'meta': {'count': 2, 'limit': 100, 'offset': 0}}

```

By type:

```
In [12]: app.content_sources.rest_client.tasks_api.list_tasks(type='snapshot')
2023-08-17 10:14:42.443 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, request_id=p0dFltTVLEOjWYrcwP7y7zOLOpntgw66, params=[('type', 'snapshot')]
Out[12]: 
{'data': [{'created_at': '2023-08-17T08:23:23Z',
           'ended_at': '2023-08-17T08:29:32Z',
           'error': '',
           'org_id': '3340851',
           'repository_name': 'fedoratest-1',
           'repository_uuid': '54e92977-9c58-43de-be31-bcacc92ce721',
           'status': 'completed',
           'type': 'snapshot',
           'uuid': '674b6996-8e8d-45b3-9532-e60b1c9257e3'}],
 'links': {'first': '/api/content-sources/v1/tasks/?limit=100&offset=0&type=snapshot',
           'last': '/api/content-sources/v1/tasks/?limit=100&offset=0&type=snapshot'},
 'meta': {'count': 1, 'limit': 100, 'offset': 0}}

In [13]: app.content_sources.rest_client.tasks_api.list_tasks(type='introspect')
2023-08-17 10:16:54.933 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, request_id=5odjHScgCyDZ4h4cuow45ZFhqpuzK8oZ, params=[('type', 'introspect')]
Out[13]: 
{'data': [{'created_at': '2023-08-17T08:23:23Z',
           'ended_at': '2023-08-17T08:29:22Z',
           'error': '',
           'org_id': '3340851',
           'repository_name': 'fedoratest-1',
           'repository_uuid': '54e92977-9c58-43de-be31-bcacc92ce721',
           'status': 'completed',
           'type': 'introspect',
           'uuid': '48bda3b4-aa25-4183-bc2e-a49009624f89'}],
 'links': {'first': '/api/content-sources/v1/tasks/?limit=100&offset=0&type=introspect',
           'last': '/api/content-sources/v1/tasks/?limit=100&offset=0&type=introspect'},
 'meta': {'count': 1, 'limit': 100, 'offset': 0}}

```

By status:

```
In [14]: app.content_sources.rest_client.tasks_api.list_tasks(status='completed')
2023-08-17 10:17:52.293 [    INFO] [iqe.base.rest_client] REST: METHOD=GET, request_id=96gTBMtJVseW6EeZVuEOjY4d6snTKs3P, params=[('status', 'completed')]
Out[14]: 
{'data': [{'created_at': '2023-08-17T08:23:23Z',
           'ended_at': '2023-08-17T08:29:22Z',
           'error': '',
           'org_id': '3340851',
           'repository_name': 'fedoratest-1',
           'repository_uuid': '54e92977-9c58-43de-be31-bcacc92ce721',
           'status': 'completed',
           'type': 'introspect',
           'uuid': '48bda3b4-aa25-4183-bc2e-a49009624f89'},
          {'created_at': '2023-08-17T08:23:23Z',
           'ended_at': '2023-08-17T08:29:32Z',
           'error': '',
           'org_id': '3340851',
           'repository_name': 'fedoratest-1',
           'repository_uuid': '54e92977-9c58-43de-be31-bcacc92ce721',
           'status': 'completed',
           'type': 'snapshot',
           'uuid': '674b6996-8e8d-45b3-9532-e60b1c9257e3'}],
 'links': {'first': '/api/content-sources/v1/tasks/?limit=100&offset=0&status=completed',
           'last': '/api/content-sources/v1/tasks/?limit=100&offset=0&status=completed'},
 'meta': {'count': 2, 'limit': 100, 'offset': 0}}
```

**Comment by swadeley** [comment]
All good.

**Pull request merged**